backoffice/backoffice-domain-service/ballerina/backoffice_impl.bal

```diff
@@ -28,32 +28,36 @@ function getAPIList() returns string?|APIList|error {
 }
 
 function changeLifeCyleState(string action, string apiId, string organization) returns LifecycleState|error {
-    string? prevLCState = check db_getCurrentLCStatus(apiId, organization);
-    string|error? lcState = db_changeLCState(action, apiId, organization);
-    if lcState is string {
-            string? newvLCState = check db_getCurrentLCStatus(apiId, organization);
-            string|error? lcEvent = db_AddLCEvent(apiId, prevLCState, newvLCState, organization);
-            if lcEvent is string {
-                json lcPayload = check getActionTransitionsFromState(action);
-                LifecycleState lcCr = check lcPayload.cloneWithType(LifecycleState);
-                return lcCr;
-            } else {
-                return error("error while adding LC event");
-            }
-    } else {
-        return error("error while updating LC state");
-    }
-} 
-
+    string prevLCState = check db_getCurrentLCStatus(apiId, organization);
+    transaction {
+        string|error lcState = db_changeLCState(action, apiId, organization);
+        if lcState is string {
+                string newvLCState = check db_getCurrentLCStatus(apiId, organization);
+                string|error lcEvent = db_AddLCEvent(apiId, prevLCState, newvLCState, organization);
+                if lcEvent is string {
+                    check commit;
+                    json lcPayload = check getActionTransitionsFromState(action);
+                    LifecycleState lcCr = check lcPayload.cloneWithType(LifecycleState);
+                    return lcCr;
+                } else {
+                    rollback;
```

Comment:
is there a possibility of using on fail error function instead of directly calling rollback, like mentioned here: https://ballerina.io/learn/by-example/mysql-atomic-transaction/

Reply:
Currently we are using custom errors. so else block return won't work unless we use rollback in every block